src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -144,34 +190,55 @@ internal Binder TypeSyntaxBinder
                 return new SourceLocalSymbol(containingSymbol, scopeBinder, allowRefKind, typeSyntax, identifierToken, declarationKind);
             }
 
-            return new LocalWithInitializer(containingSymbol, scopeBinder, typeSyntax, identifierToken, initializer, initializerBinderOpt, declarationKind);
+            return new LocalWithInitializer(containingSymbol, scopeBinder, typeSyntax, identifierToken, initializer, initializerBinderOpt ?? scopeBinder, declarationKind);
         }
 
+        /// <summary>
+        /// Make a local variable for an out variable declaration whose type is inferred as a side-effect of binding the enclosing expression.
+        /// </summary>
         /// <param name="containingSymbol"></param>
-        /// <param name="scopeBinder">
-        /// Binder that owns the scope for the local, the one that returns it in its <see cref="Binder.Locals"/> array.
-        /// </param>
-        /// <param name="enclosingBinderOpt">
-        /// Enclosing binder for the location where the local is declared, if different from the <paramref name="scopeBinder"/>.
-        /// It should be used to bind something at that location.
-        /// </param>
-        /// <param name="typeSyntax"></param>
+        /// <param name="scopeBinder">The binder for the scope of the local</param>
+        /// <param name="invocationBinderOpt">The binder for the invocation, if different from scopeBinder</param>
+        /// <param name="typeSyntax">The type syntax</param>
         /// <param name="identifierToken"></param>
-        /// <param name="context"></param>
-        public static SourceLocalSymbol MakeOutVariable(
+        /// <param name="declarationKind"></param>
+        /// <param name="context">The expression to be bound, which will result in the variable receiving a type</param>
+        /// <returns></returns>
+        public static SourceLocalSymbol MakeVariableDeclaredInExpression(
```

Comment:
>MakeVariableDeclaredInExpression [](start = 40, length = 32)

The doc comment says this function is specific for "out" locals, but the name doesn't reflect this. Is there a particular reason for this?